interface.go

```diff
@@ -27,6 +28,63 @@ func NewDefault(title string, l ...level) *logger {
 	return New(os.Stdout, title, ShortCaller, lvl, DefaultSeparator)
 }
 
+type Config struct {
+	Title             string `yaml:"title"`
+	Separator         string `yaml:"separator"`
+	Level             level  `yaml:"level"`
+	Direction         string `yaml:"direction"`
+	EnableDate        bool   `yaml:"enable_date"`
+	EnableTime        bool   `yaml:"enable_time"`
+	EnableLabels      bool   `yaml:"enable_labels"`
+	EnableCaller      bool   `yaml:"enable_caller"`
+	EnableShortCaller bool   `yaml:"enable_short_caller"`
+}
+
+func NewFromConfig(cfg Config) *logger {
+	l := new(logger)
+	l.title = cfg.Title
+
+	if cfg.Direction == "stdout" {
+		l.SetWriter(os.Stdout)
+	} else if cfg.Direction == "stderr" {
+		l.SetWriter(os.Stderr)
+	} else if cfg.Direction == "" {
+		l.SetWriter(os.Stdout)
+	} else {
+		f, _ := os.OpenFile(cfg.Direction, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+		l.SetWriter(f)
+	}
+	if cfg.Level == "" {
+		l.SetLevel(WARNING)
+	} else {
+		l.SetLevel(level(strings.ToUpper(string(cfg.Level))))
```

Comment:
what will happen if we set `asd` as level? we need to check for validity of specified value and have a test for invalid cases